solidity/contracts/MezoAllocator.sol

```diff
@@ -225,20 +231,32 @@ contract MezoAllocator is IDispatcher, Ownable2StepUpgradeable {
     function withdraw(uint256 amount) external {
         if (msg.sender != address(stbtc)) revert CallerNotStbtc();
 
-        emit DepositWithdrawn(depositId, amount);
+        uint256 unallocatedBalance = tbtc.balanceOf(address(this));
+
+        if (amount > unallocatedBalance) {
+            uint256 amountToWithdraw = amount - unallocatedBalance;
+
+            emit DepositWithdrawn(depositId, amountToWithdraw);
+
+            if (amountToWithdraw < depositBalance) {
```

Comment:
How about changing the order of the logic?
```
 if (amountToWithdraw < depositBalance) {
            mezoPortal.withdrawPartially(
                address(tbtc),
                depositId,
                uint96(amountToWithdraw)
            );
        } else if (amountToWithdraw == depositBalance) {
            mezoPortal.withdraw(address(tbtc), depositId);
        } else {
            revert WithdrawalAmountExceedsDepositBalance(
                amountToWithdraw,
                depositBalance
            );
        }
```

```
if(..) {
  valid case
} else if (..) {
  valid case
} else {
  error
}
```

instead of error being in the middle. Might be somewhat cleaner. Up to you, not a blocker.

Reply:
7b97aac7f45a59b73a9e73f505e3d9e1aa1f922a